app/src/main/java/sopt/org/millie/core/designsystem/component/SearchTextField.kt

```diff
@@ -0,0 +1,104 @@
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import sopt.org.millie.R
+import sopt.org.millie.core.designsystem.theme.MillieTheme
+import sopt.org.millie.core.util.noRippleClickable
+
+@Composable
+fun SearchTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
+    trailingIcon: @Composable () -> Unit = {},
+    onSearchAction: () -> Unit = {},
+) {
+    val keyboardController = LocalSoftwareKeyboardController.current
+
+    TextField(
+        value = value,
+        onValueChange = onValueChange,
+        modifier = modifier
+            .fillMaxWidth()
+            .clip(RoundedCornerShape(22.dp)),
+        textStyle = MillieTheme.typography.body.body2,
+        placeholder = {
+            Text(
+                text = "검색어를 입력하세요",
+                color = MillieTheme.colors.gray3,
+                style = MillieTheme.typography.body.body2,
+            )
+        },
+        leadingIcon = {
+            Icon(
+                imageVector = ImageVector.vectorResource(R.drawable.ic_search),
+                contentDescription = "검색",
+                tint = MillieTheme.colors.gray3,
+                modifier = Modifier.size(24.dp),
+            )
+        },
+        trailingIcon = {
+            if (value.isNotEmpty()) {
+                Icon(
+                    imageVector = ImageVector.vectorResource(R.drawable.ic_search_close),
+                    contentDescription = "삭제",
+                    tint = Color.Unspecified,
+                    modifier = Modifier
+                        .size(24.dp)
+                        .noRippleClickable(
+                            onClick = { onValueChange("") },
+                        ),
+                )
+            } else {
+                trailingIcon()
+            }
+        },
+        keyboardOptions = KeyboardOptions(
+            imeAction = ImeAction.Search,
+        ),
+        keyboardActions = KeyboardActions(
+            onDone = {
+                onSearchAction()
+                keyboardController?.hide()
+            },
+            onSearch = {
+                onSearchAction()
+                keyboardController?.hide()
+            },
+        ),
+        singleLine = true,
+        colors = TextFieldDefaults.colors(
+            focusedContainerColor = MillieTheme.colors.lightGray1,
+            unfocusedContainerColor = MillieTheme.colors.lightGray1,
+            focusedIndicatorColor = Color.Transparent,
+            unfocusedIndicatorColor = Color.Transparent,
+        ),
+    )
+}
+
+@Preview(showBackground = true)
+@Composable
+private fun SearchTextFieldPreview() {
+    SearchTextField(
+        value = "",
+        onValueChange = {},
+    )
+}
```

Comment:
```suggestion
@Preview(showBackground = true)
@Composable
private fun SearchTextFieldPreview() {
    MillieTheme {
        var text by remember { mutableStateOf("") }

        SearchTextField(
            value = text,
            onValueChange = { text = it },
        )
    }
}
```
요거 MillieTheme{} 로 안감싸주면 컬러랑 타이포를 찾을수가 없어서 프리뷰 오류날 가능성이 클텐데 ...!?!?! 그리구 interactionmode에서 타자 쳐지는거 확인해보려면 이렇게 작성하는건 어떨까요 ~ 히히

Reply:
헉 이런 방법이... mainactivity에 몰래 추가해서 실행해봤어요....ㅎㅎ

요렇게 수정해볼게요!!

Reply:
귀엽다....ㅎ.ㅎ